srml/contract/src/tests.rs

```diff
@@ -119,19 +120,22 @@ impl ContractAddressFor<H256, u64> for DummyContractAddressFor {
 	}
 }
 
-static KEY_COUNTER: AtomicUsize = AtomicUsize::new(0);
-
 pub struct DummyTrieIdGenerator;
 impl TrieIdGenerator<u64> for DummyTrieIdGenerator {
```

Comment:
Could we remove this struct and use the actual TrieIdGenerator? It feels to me like it is the same code (I may have miss something).

Reply:
well TrieIdGenerator in src/lib.rs requires T::Account to be `As<[u8]>` which is not the case of u64.

but we could be able to manage this as I said in https://github.com/paritytech/substrate/pull/2502#issuecomment-490897102